src/core/build_label.go

```diff
@@ -303,6 +314,40 @@ func (label BuildLabel) IsEmpty() bool {
 	return label.PackageName == "" && label.Name == ""
 }
 
+// toArch returns a new build label that's a copy of this one with a new architecture.
+func (label BuildLabel) toArch(arch string) BuildLabel {
+	label.Arch = arch
+	return label
+}
+
+// noArch returns a new build label with no architecture.
+func (label BuildLabel) noArch() BuildLabel {
+	label.Arch = ""
+	return label
+}
+
+// FullArch returns the full architecture, even if Arch is empty, in which case it will return
+// the host architecture.
+func (label BuildLabel) FullArch() string {
```

Comment:
Done, this type is pretty old and for mostly bad reasons generally doesn't use pointer receivers (in fact I generally pass them by value). Sometime I'll change it all over although it's quite a wide-ranging change so I'll leave it out of this PR.